### PR TITLE
Version Packages

### PR DESCRIPTION
**Pull request by hashibot-hds**
This PR was opened by the [Changesets release](https://github.com/changesets/action) GitHub action. When you're ready to do a release, you can merge this and the packages will be published to npm automatically. If you're not ready to do a release yet, that's fine, whenever you add more changesets to main, this PR will be updated.

# Releases
### Major Changes

Converted Ember packages to v2 addon format.

To migrate update Sass cofiguration in `ember-cli-build.js` to include the paths for `ember-flight-icons` and `design-system-components`:

```js
sassOptions: {
  precision: 4,
  includePaths: [
    './node_modules/@hashicorp/design-system-tokens/dist/products/css',
    './node_modules/@hashicorp/ember-flight-icons/dist/styles'
    './node_modules/@hashicorp/design-system-components/dist/styles',
  ],
},
```

Alternatively, you can import the CSS by adding this configuration in `ember-cli-build.js`.

```js
app.import(
  "node_modules/@hashicorp/design-system-components/dist/styles/@hashicorp/design-system-components.css"
);
```

<small class="doc-whats-new-changelog-metadata">[#1872](https://github.com/hashicorp/design-system/pull/1872)</small>

<div class="doc-whats-new-changelog-separator"></div>

`Form::CharacterCount` - refactored the component, removing `onInsert` callback and adding use `@value` argument

To migrate:

-   for standalone `Form::CharacterCount` instances you must pass in a `@value` arguments representing the value of the referenced input
-   when used as a contextual component `F.CharacterCount` in `Form::[MaskedInput|TextInput|Textarea]::Field` make sure the form control is updating the associated `@value` on input (usually using `{{on "input" (fn this.updateValue)}}` function)

<small class="doc-whats-new-changelog-metadata">[#1896](https://github.com/hashicorp/design-system/pull/1896) - Thanks [@meirish](https://github.com/meirish) for the contribution! 🙏</small>

<div class="doc-whats-new-changelog-separator"></div>

`Table` - Multiple updates to the main component and its subcomponents:

-   Updated table headers to support tooltips
-   Updated visual treatment and location of the "sort" button in the table headers
-   Refactored CSS code to simplify usage of `hds-table`-related class names

`Table::ThSort`:

-   Added support for tooltip via the `@tooltip` argument
-   Updated visual treatment and location of the "sort" button
-   Updated DOM structure of the `<th>` content
-   Remove class `hds-table__th-sort--button-content`
-   Replaced class `hds-table__th-sort` with classes `hds-table__th` + `hds-table__th--sort`
-   Replaced class `hds-table__th-sort--text--[left|center|right]` with `hds-table__th--align-[left|center|right]`
-   Renamed `onClick` callback to `onClickSort`

`Table::Th`:

-   Added support for tooltip via the `@tooltip` argument
-   Updated DOM structure of the `<th>` content
-   Replaced class `hds-table__th--text-[left|center|right]` with `hds-table__th--align-[left|center|right]`

`Table::Td`:

-   Replaced class `hds-table__td--text-[left|center|right]` with `hds-table__td--align-[left|center|right]`

<small class="doc-whats-new-changelog-metadata">[#1860](https://github.com/hashicorp/design-system/pull/1860)</small>

<div class="doc-whats-new-changelog-separator"></div>

`Pagination` - Removed handling of query parameters from `onPageSizeChange` function for `Pagination::Numbered`.

_Unfortunately, it's not possible to cover this breaking change with a codemod. Consumers should review their usage of the `onPageSizeChange` callback and, if necessary, implement the persistence of the "page number" and "page size" values via query parameters themselves._

<small class="doc-whats-new-changelog-metadata">[#1913](https://github.com/hashicorp/design-system/pull/1913)</small>

<div class="doc-whats-new-changelog-separator"></div>

Renamed namespaced contextual components as follows:

-   `Alert::Link::Standalone` to `Alert::LinkStandalone`
-   `ApplicationState::Footer::Link::Standalone` to `ApplicationState::Footer::LinkStandalone`
-   `Form::Checkbox::Group::Checkbox::Field` to `Checkbox::Group::CheckboxField`
-   `Form::Radio::Group::Radio::Field` to `Form::Radio::Group::RadioField`
-   `Form::Toggle::Group::Toggle::Field` to `Form::Toggle::Group::ToggleField`
-   `Toast::Link::Standalone` to `Toast::LinkStandalone`

<small class="doc-whats-new-changelog-metadata">[#1884](https://github.com/hashicorp/design-system/pull/1884)</small>

<div class="doc-whats-new-changelog-separator"></div>

### Minor Changes

`Table` - Added multi-select functionality

<small class="doc-whats-new-changelog-metadata">[#1859](https://github.com/hashicorp/design-system/pull/1859)</small>

<div class="doc-whats-new-changelog-separator"></div>

`Tabs` - Added `@size` argument with new "large" size variant

<small class="doc-whats-new-changelog-metadata">[#1937](https://github.com/hashicorp/design-system/pull/1937)</small>

<div class="doc-whats-new-changelog-separator"></div>

### Patch Changes

`Dropdown` - Fixed dropdown list missing an accessible name when Checkmark items were passed in

<small class="doc-whats-new-changelog-metadata">[#1910](https://github.com/hashicorp/design-system/pull/1910)</small>

<div class="doc-whats-new-changelog-separator"></div>

`Flyout` - Reduce gap between Flyout and edge of screen from 40px to half of the minimized `SideNav` width in medium view.

<small class="doc-whats-new-changelog-metadata">[#1957](https://github.com/hashicorp/design-system/pull/1957)</small>

<div class="doc-whats-new-changelog-separator"></div>

Remove `ember-deep-tracked` dependency that was not used

<small class="doc-whats-new-changelog-metadata">[#1950](https://github.com/hashicorp/design-system/pull/1950)</small>

<div class="doc-whats-new-changelog-separator"></div>

`SideNav` - fixed issue with navigation elements remaining interactive when minimized

<small class="doc-whats-new-changelog-metadata">[#1909](https://github.com/hashicorp/design-system/pull/1909)</small>

<div class="doc-whats-new-changelog-separator"></div>

**🔄 Updated dependencies:**

-   @hashicorp/ember-flight-icons@5.0.0
-   @hashicorp/design-system-tokens@2.0.0

### Major Changes

Converted Ember packages to v2 addon format.

To migrate update Sass cofiguration in `ember-cli-build.js` to include the paths for `ember-flight-icons` and `design-system-components`:

```js
sassOptions: {
  precision: 4,
  includePaths: [
    './node_modules/@hashicorp/design-system-tokens/dist/products/css',
    './node_modules/@hashicorp/ember-flight-icons/dist/styles'
    './node_modules/@hashicorp/design-system-components/dist/styles',
  ],
},
```

Alternatively, you can import the CSS by adding this configuration in `ember-cli-build.js`.

```js
app.import(
  "node_modules/@hashicorp/design-system-components/dist/styles/@hashicorp/design-system-components.css"
);
```

<small class="doc-whats-new-changelog-metadata">[#1872](https://github.com/hashicorp/design-system/pull/1872)</small>

<div class="doc-whats-new-changelog-separator"></div>

### Major Changes

New tokens added for large and medium size `Tab` variants. The "medium" tokens replace the former equivalent tokens which were not differentiated by size.

-   `--token-tabs-tab-height-medium` (**Replaces `--token-tabs-tab-height`**)
-   `--token-tabs-tab-height-large` (**New**)
-   `--token-tabs-tab-padding-horizontal-medium` (**Replaces `--token-tabs-tab-padding-horizontal`**)
-   `--token-tabs-tab-padding-horizontal-large` (**New**)

<small class="doc-whats-new-changelog-metadata">[#1937](https://github.com/hashicorp/design-system/pull/1937)</small>

<div class="doc-whats-new-changelog-separator"></div>

### Patch Changes

Add codemods for `design-system-components` v4

<small class="doc-whats-new-changelog-metadata">[#1884](https://github.com/hashicorp/design-system/pull/1884)</small>

<div class="doc-whats-new-changelog-separator"></div>

### CODE REVIEW

**Comment by didoo** [comment]
@alex-ju before merging, let's add a `how to migrate` also for the table's breaking changes (and the renaming of the subcomponents, if you haven't yet added it to the changeset file

File: packages/codemods/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,15 @@
 # @hashicorp/design-system-codemods
 
+## 0.1.1
+
+### Patch Changes
+
+Add codemods for `design-system-components` v4
+
+<small class="doc-whats-new-changelog-metadata">[#1884](https://github.com/hashicorp/design-system/pull/1884)</small>
```
Comment (didoo) [inline packages/codemods/CHANGELOG.md:9]:
```suggestion
<small class="doc-whats-new-changelog-metadata">[#1884](https://github.com/hashicorp/design-system/pull/1884) / [#1860](https://github.com/hashicorp/design-system/pull/1860)</small>
```

File: packages/components/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,104 @@
 # @hashicorp/design-system-components
 
+## 4.0.0
+
+### Major Changes
+
+Converted Ember packages to v2 addon format.
+
+To migrate update Sass cofiguration in `ember-cli-build.js` to include the paths for `ember-flight-icons` and `design-system-components`:
```
Comment (didoo) [inline packages/components/CHANGELOG.md:9]:
```suggestion
To migrate update Sass configuration in `ember-cli-build.js` to include the paths for `ember-flight-icons` and `design-system-components`:
```

File: packages/ember-flight-icons/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,36 @@
 # @hashicorp/ember-flight-icons
 
+## 5.0.0
+
+### Major Changes
+
+Converted Ember packages to v2 addon format.
+
+To migrate update Sass cofiguration in `ember-cli-build.js` to include the paths for `ember-flight-icons` and `design-system-components`:
```
Comment (didoo) [inline packages/ember-flight-icons/CHANGELOG.md:9]:
```suggestion
To migrate update Sass configuration in `ember-cli-build.js` to include the paths for `ember-flight-icons` and `design-system-components`:
```

File: website/docs/whats-new/release-notes/partials/components.md [hunk 4]
```diff
@@ -12,6 +12,105 @@
   </a>
 </p>
 
+## 4.0.0
+
+**Major changes**
+
+Converted Ember packages to v2 addon format.
+
+To migrate update Sass cofiguration in `ember-cli-build.js` to include the paths for `ember-flight-icons` and `design-system-components`:
```
Comment (didoo) [inline website/docs/whats-new/release-notes/partials/components.md:21]:
```suggestion
To migrate update Sass configuration in `ember-cli-build.js` to include the paths for `ember-flight-icons` and `design-system-components`:
```

File: website/docs/whats-new/release-notes/partials/ember-flight-icons.md [hunk 5]
```diff
@@ -12,6 +12,37 @@
   </a>
 </p>
 
+## 5.0.0
+
+**Major changes**
+
+Converted Ember packages to v2 addon format.
+
+To migrate update Sass cofiguration in `ember-cli-build.js` to include the paths for `ember-flight-icons` and `design-system-components`:
```
Comment (didoo) [inline website/docs/whats-new/release-notes/partials/ember-flight-icons.md:21]:
```suggestion
To migrate update Sass configuration in `ember-cli-build.js` to include the paths for `ember-flight-icons` and `design-system-components`:
```

File: packages/components/CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,104 @@
...
+`Table::Th`:
+
+- Added support for tooltip via the `@tooltip` argument
+- Updated DOM structure of the `<th>` content
+- Replaced class `hds-table__th--text-[left|center|right]` with `hds-table__th--align-[left|center|right]`
+
+`Table::Td`:
+
+- Replaced class `hds-table__td--text-[left|center|right]` with `hds-table__td--align-[left|center|right]`
+
```
Comment (didoo) [inline packages/components/CHANGELOG.md:70]:
```suggestion

To migrate run the codemod `v4/table` (see [readme file](https://github.com/hashicorp/design-system/tree/main/packages/codemods/transforms/v4/table))

```

File: website/docs/whats-new/release-notes/partials/components.md [hunk 7]
```diff
@@ -12,6 +12,105 @@
...
+`Table::Th`:
+
+- Added support for tooltip via the `@tooltip` argument
+- Updated DOM structure of the `<th>` content
+- Replaced class `hds-table__th--text-[left|center|right]` with `hds-table__th--align-[left|center|right]`
+
+`Table::Td`:
+
+- Replaced class `hds-table__td--text-[left|center|right]` with `hds-table__td--align-[left|center|right]`
+
```
Comment (didoo) [inline website/docs/whats-new/release-notes/partials/components.md:82]:
```suggestion

To migrate run the codemod `v4/table` (see [readme file](https://github.com/hashicorp/design-system/tree/main/packages/codemods/transforms/v4/table))

```

**Review by didoo — COMMENTED** [review]
@alex-ju left a few suggestions

File: packages/codemods/package.json [hunk 8]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@hashicorp/design-system-codemods",
   "description": "Helios Design System codemods",
-  "version": "0.1.0",
+  "version": "0.1.1",
```
Comment (alex-ju) [inline packages/codemods/package.json:4]:
```suggestion
  "version": "0.2.0",
```

File: packages/codemods/CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,15 @@
 # @hashicorp/design-system-codemods
 
+## 0.1.1
```
Comment (alex-ju) [inline packages/codemods/CHANGELOG.md:3]:
```suggestion
## 0.2.0
```

File: packages/codemods/CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,15 @@
...
+
+### Patch Changes
```
Comment (alex-ju) [inline packages/codemods/CHANGELOG.md:5]:
```suggestion
### Minor Changes
```

**Review by alex-ju — APPROVED** [review]

**Pull request merged**